packages/store-indexer/bin/postgres-decoded-indexer.ts

```diff
@@ -35,16 +37,37 @@ const publicClient = createPublicClient({
 });
 
 const chainId = await publicClient.getChainId();
-const database = drizzle(postgres(env.DATABASE_URL, { prepare: false }));
 
-const { storageAdapter, tables } = await createStorageAdapter({ database, publicClient });
+let readDatabaseUrl = env.READ_DATABASE_URL;
+let writeDatabaseUrl = env.WRITE_DATABASE_URL;
+
+if (env.DATABASE_URL) {
+  console.warn("DATABASE_URL is deprecated. Please use READ_DATABASE_URL and WRITE_DATABASE_URL");
```

Comment:
it doesn't have to be deprecated imo, we could just say if you want both to be the same you can use `DATABASE_URL` (which I would expect to be a common use case too)

With that we can simplify this whole logic to:

```
const readDatabaseUrl = env.READ_DATABASE_URL ?? env.DATABASE_URL;
const writeDatabaseUrl = env.WRITE_DATABASE_URL ?? env.DATABASE_URL;

if(!readDatabaseUrl || !writeDatabaseUrl) {
  throw new Error("No database URL defined. Please provide either `DATABASE_URL` or `READ_DATABASE_URL` and `WRITE_DATABASE_URL` environment variables.");
}